contracts/MoMABurn.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract FFV2 {
+    struct FF2Artwork {
+        string title;
+        string artistName;
+        string fingerprint;
+        uint256 editionSize;
+    }
+
+    mapping(uint256 => FF2Artwork) public artworks;
+
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public {}
+}
+
+contract FFV33 {
+    struct FF3Artwork {
+        string title;
+        string artistName;
+        string fingerprint;
+        uint256 editionSize;
+        uint256 AEAmount;
+        uint256 PPAmount;
+    }
+    mapping(uint256 => FF3Artwork) public artworks;
+
+    function mintArtworkEdition(uint256 _artworkID, address _owner) public {}
+}
+
+contract MoMABurn is Ownable {
+    using SafeMath for uint256;
+    address constant DEAD_ADDRESS = 0x000000000000000000000000000000000000dEaD;
+
+    address public burnContractAddress; // Address of token need to burn
+    address public tier1ContractAddress; // Address of new token for Tier 1
+    address public tier2ContractAddress; // Address of new token for Tier 2
+    address public tier3ContractAddress; // Address of new token for Tier 3
+
+    uint256 private burnArtworkID; // Artwork ID will be burned
+    uint256 private tier1ArtworkID; // Artwork ID of Tier 1
+    uint256 private tier2ArtworkID; // Artwork ID of Tier 2
+    uint256 private tier3ArtworkID; // Artwork ID of Tier 3
+
+    bool public isBurnEnabled;
+
+    constructor() {
+        isBurnEnabled = false;
+    }
+
+    /// @notice setBurnAndMintParams use for update burn and mint contract addresses and artwork IDs
+    /// @param _burnContractAddress - new burn contract address
+    /// @param _tier1Address - new burn contract address for tier 1
+    /// @param _tier2Address - new burn contract address for tier 2
+    /// @param _tier3Address - new burn contract address for tier 3
+    /// @param _burnArtworkID - artwork ID will be burned
+    /// @param _tier1ArtworkID - new artwork ID for tier 1
+    /// @param _tier2ArtworkID - new artwork ID for tier 2
+    /// @param _tier3ArtworkID - new artwork ID for tier 3
+    function setBurnAndMintParams(
+        address _burnContractAddress,
+        address _tier1Address,
+        address _tier2Address,
+        address _tier3Address,
+        uint256 _burnArtworkID,
+        uint256 _tier1ArtworkID,
+        uint256 _tier2ArtworkID,
+        uint256 _tier3ArtworkID
+    ) public onlyOwner {
+        require(
+            _burnContractAddress != address(0) &&
+                _tier1Address != address(0) &&
+                _tier2Address != address(0) &&
+                _tier3Address != address(0),
+            "Invalid contract address"
+        );
+        (, , string memory burnFingerprint, uint256 burnEditionSize) = FFV2(
+            _burnContractAddress
+        ).artworks(_burnArtworkID);
```

Comment:
Why not declare to variable `burnArtwork` directly? It makes the code hard to read. Do we use this way because we want to save the cost?

Reply:
let's create a private function for v2 and v3. For example, 
`function getV2ContractArtwork(address, artworkID) returns (uint256, string)`

So this ugly thing would only in a place.